adding changes to GA4 topics (#6262)
segmentio/segment-docs: src/connections/destinations/catalog/actions-google-analytics-4-web/index.md

> info "Consent mode"
> Google enforced consent on March 6, 2024 for European Economic Area (EEA) users. Learn more about [consent mode](/docs/connections/destinations/catalog/actions-google-analytics-4-web/#consent-mode) and how to set it up. 

## Getting started

Before you connect Segment to Google Analytics 4, configure a Google Analytics 4 property in your Analytics account. Additionally, plan and enable any Custom Definitions in your GA4 Admin Panel prior to sending data from Segment to GA4. For more information, see Google's article [Set up Analytics for a website and/or app](https://support.google.com/analytics/answer/9304153){:target='_blank'}.

To connect the Google Analytics 4 Web destination:

1. From the Segment web app, click **Catalog**, then click **Destinations**.
2. Search for “Google Analytics 4 Web” in the Destinations Catalog, and select the destination.
3. Click **Configure Google Analytics 4 Web**.

> 4. Select the web source that will send data to Google Analytics 4 and follow the steps to name your destination. The web source chosen must use [Analytics.js 2.0](/docs/connections/sources/catalog/libraries/website/javascript/). For mobile source tracking, view Firebase Destination configuration {{insert link}}.

Reviewer: insert link to Segment Firebase Destination

8. For GA4 to accept events on page, enable Set Configuration Mapping triggered by the first Segment event called after analytics.load(). Set Configuration Mapping calls the gtag(‘config’) command to enable tracking to your GA4 Measurement ID. This command MUST be called at each new page load.

After you've set up and enabled the Set Configuration Mapping, enable at least one event in your **Mappings** tab. From there, view your events and parameters using the Google [Realtime](https://support.google.com/analytics/answer/9271392){:target="_blank"} or[DebugView](https://support.google.com/analytics/answer/7201382){:target="_blank"} reports. These two reports show you the events users trigger on your website as they occur. The DebugView report requires additional configuration before you can use it. Additional tools for debugging are to view all your Google enabled tracking via https://tagassistant.google.com(https://tagassistant.google.com){:target=”_blank”} or in your browser’s Dev Tools, view GA4 collect requests by filtering by /collect.

> Some events automatically populate in Google Analytics. For example, there are some [Automatically Collected] (https://support.google.com/analytics/answer/9234069){:target=”_blank”} events collected by triggering the Set Configuration Mapping. For example, calling gtag(‘config’), and [Enhanced Measurement events](https://support.google.com/analytics/answer/9216061){:target=”_blank”} controlled by toggling “on” in your GA4 Admin panel use event listeners to send events; in addition, all events tracked via GA4 Web, populate some commonly used parameters like page_location without additional configuration. Review Google’s documents for more information found [here](https://support.google.com/analytics/table/13594742){:target=”_blank”}

Reviewer: Edit using markdown to highlight, do not add ticks to final:
`Some events automatically populate in Google Analytics. For example, there are some [Automatically Collected] (https://support.google.com/analytics/answer/9234069){:target=”_blank”} events collected by triggering the Set Configuration Mapping. Additionally, calling gtag(‘config’), and enabling [Enhanced Measurement events](https://support.google.com/analytics/answer/9216061){:target=”_blank”}, which are controlled by toggling “on” in your GA4 Admin panel, will use event listeners to send events; moreover, all events tracked via GA4 Web, populate some commonly used parameters like page_location without additional configuration. Review Google’s documents for more information found [here](https://support.google.com/analytics/table/13594742){:target=”_blank”}`

### Recommended events

> Google Analytics 4 advises for the use of [Recommended events](https://support.google.com/analytics/answer/9267735){:target="_blank"} and properties to power certain built-in reports. Segment’s Google Analytics 4 Web destination provides prebuilt mappings to automatically map your Segment spec events to the corresponding Google Analytics 4 events and properties. If your Segment events don’t follow the Segment spec exactly, you can modify the mappings. For example, Segment maps `Order Completed` events to the Google Analytics 4 `Purchase` event by default. If your company uses `Products Purchase` to indicate a purchase, you can map it in the Purchase action’s Event Trigger instead.

Author: ```suggestion
Google Analytics 4 advises for the use of [Recommended events](https://support.google.com/analytics/answer/9267735){:target="_blank"} and properties to power certain built-in reports. Segment’s Google Analytics 4 Web destination provides prebuilt mappings to automatically map your Segment spec events to the corresponding Google Analytics 4 events and properties. If your Segment events don’t follow the Segment spec exactly, you can modify the mappings. or example, Segment maps `Order Completed` events to the Google Analytics 4 `Purchase` event when you add the `Purchase` action to your mappings. If your company uses `Products Purchase` to indicate a purchase, you can map it in the Purchase action’s Event Trigger instead.
```

To enable consent mode for your Google Analytics 4 Web destination: 
1. Navigate to **Connections > Destinations** and select your Google Analytics 4 Web** destination. 
2. Go to the **Settings** tab of the destination.

> 3. Click the toggle on for **Enable Consent Mode**, this calls gtag(‘consent’,’default’) with the defined parameters when the gtag library loads.

Reviewer: I added a note not to modify this section.  I'm not confident in my knowledge of what changes were made; therefore, remove any edits from Consent Mode section.

> 4. Set the following fields with your organization’s determination of granted or denied:

Reviewer: Remove change

    Field | Value
    ----- | ------
    Default Ad Storage Consent State | Granted or Denied
    Default Analytics Storage Consent State | Granted or Denied
    Ad User Data Consent State | Granted or Denied

>     Ad Personalization Consent State | Granted or Denied

Reviewer: Remove changes on 92-95

        'Analytics Storage Consent State': 'false'
    });
    ```

> 7. As soon as the page loads and the set configuration fires to the Google Analytics SDK, Segment issues a consent mode update command. Map the properties you defined to collect consent state changes to the Set Configurations Fields mapping. You can choose to do this from 1 of 2 options in steps 4 and 5.

Reviewer: Remove

            * Ad User Data Consent State
            * Ad Personalization Consent State
            You can manually select `Granted` or `Denied` from the dropdown menu for Advanced consent mode settings, and type in `granted` or `denied` for basic consent mode settings.

>         5. *(Option 2)* Under the **Select mappings** section, create an event variable to directly grab the value from the payload. For example, `properties.adStorageConsentState`. Ensure it translates to `granted` or `denied`. You can use an insert function to translate to `granted` or `denied`. Use the replace function if it's a string. Do this for these fields:

Reviewer: remove

            * Ads Storage Consent State
            * Analytics Storage Consent State
            * Ad User Data Consent State
            * Ad Personalization Consent State

> When these properties are available, they send to the `update` command.

Reviewer: remove
